### ISSUE TITLE
there are some difference and misunderstanding in your code

### ISSUE THREAD

**Issue by MrDruan** [Riathoir/PASSGAN-IWGAN-Tensorflow-2#2]
First , in the original paper ,  **Z** is a 128-dimensional vector generated by tf.random.normal (without regard to the batchsize ), then transmit it into the full connection layer . **The input dim of the full connection layer** is 128 and the output dim is 120*seq_len . 

But in youe code , **Z** is [2,1,32] ?(I dont know why) , and the full connection layer in your code is also different from the original code because  the use of keras . In your code , **The input dim of the full connection layer** is 128*seq_len and the output dim is 128 , whitch is totally different from the core idea of the paper . I am a new learner in PassGAN , I wonder if  there is some different ideas or you have taken account of other conditions in  your source code? 

**Looking forward to your reply**

**Comment by Riathoir** [comment]
Thanks for taking the time to review this repository and raise this issue, I must apologise for the length of time it has taken for me to respond as I no longer monitor this account. 
I believe you are misunderstanding the paper and the original source code it was based off, may I refer you to the excellent work [here](https://github.com/igul222/improved_wgan_training/blob/master/gan_language.py)

The PassGAN paper is, when speaking candidly, a clone of this codebase with minor tweaks. PassGAN is a language IWGAN  authored [here](https://arxiv.org/pdf/1704.00028.pdf) and takes little more than a different dataset approach to self-supervised language learning. Visualisation of PhD work is at times difficult (PassGAN), so I would recommend you read the above instead. 

The README to this repository is marginally misleading without the accompanying paper, which I will happily forward to you if you would like, as it was only ever distributed internally. 
The brief summary is **yes** there are several design differences to this code base than the original, hence it is and was unsuitable to use as validation. For that, I could have simply cloned the repository above and applied the changes. 
This repository humorously improves PassGANs attempt, with this baseline being made public and the second generation remaining private due to well founded concerns over responsibility, as discussed in the paper - this first generation repository in its current state is not of concern.  

If you have any further queries please do reach out. Have a lovely St Paddies day. 
Rachel

**Comment by MrDruan** [comment]
thank you very much Racheal，I would like to read the literature you offered for me and I would like to discuss with you about academic. I am from China,Please forgive my poor English, I feel so happy today because you replied me . Happy St Paddies Day and have a good time 🤞🤞🤞🥰🥰🥰发自我的iPhone------------------ Original ------------------From: Rachel  ***@***.***>Date: Wed,Mar 17,2021 10:05 PMTo: Riathoir/PASSGAN-IWGAN-Tensorflow-2 ***@***.***>Cc: MrDruan ***@***.***>, Author ***@***.***>Subject: Re: [Riathoir/PASSGAN-IWGAN-Tensorflow-2] there are some differenceand misunderstanding in your code (#2)
Thanks for taking the time to review this repository and raise this issue, I must apologise for the length of time it has taken for me to respond as I no longer monitor this account.
I believe you are misunderstanding the paper and the original source code it was based off, may I refer you to the excellent work here
The PassGAN paper is, when speaking candidly, a clone of this codebase with minor tweaks. PassGAN is a language IWGAN  authored here and takes little more than a different dataset approach to self-supervised language learning. Visualisation of PhD work is at times difficult (PassGAN), so I would recommend you read the above instead.
The README to this repository is marginally misleading without the accompanying paper, which I will happily forward to you if you would like, as it was only ever distributed internally.
The brief summary is yes there are several design differences to this code base than the original, hence it is and was unsuitable to use as validation. For that, I could have simply cloned the repository above and applied the changes.
This repository humorously improves PassGANs attempt, with this baseline being made public and the second generation remaining private due to well founded concerns over responsibility, as discussed in the paper - this first generation repository in its current state is not of concern.
If you have any further queries please do reach out. Have a lovely St Paddies day.
Rachel

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/Riathoir/PASSGAN-IWGAN-Tensorflow-2/issues/2#issuecomment-801108170",
"url": "https://github.com/Riathoir/PASSGAN-IWGAN-Tensorflow-2/issues/2#issuecomment-801108170",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]
